core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLPrettyWriterTest.java

```diff
@@ -104,6 +109,58 @@ public void outOfSequenceItemsAreNotAbbreviated() throws RDFHandlerException, IO
 		assertEquals(Arrays.asList("<rdf:RDF", "<rdf:Bag", "<rdf:_0", "<rdf:_2"), rdfLines);
 	}
 
+	@Test
+	public void compactXMLPrintTest() throws RDFHandlerException, IOException, ClassNotFoundException,
+			NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+		OutputStream outputStream = new ByteArrayOutputStream();
+		PrintStream printStream = new PrintStream(outputStream);
+		System.setOut(printStream);
```

Comment:
Rather than redirecting System.out, why not just give the outputStream argument directly to the RDFXMlPrettyWriter?